File: web-api/ControleEstoque.Web/Models/Domain/Usuario.cs
using Dapper;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Configuration;
using System.Data.SqlClient;
using System.Linq;
using ControleEstoque.Web.Models.Domain;
namespace ControleEstoque.Web.Models
{
    public class Usuario
    {
 
        public int Id { get; set; }
        [Required(ErrorMessage = "Informe o login")]
        [MaxLength(50, ErrorMessage = "O login pode ter no máximo 50 caracteres.")]
        public string Login { get; set; }
        [Required(ErrorMessage = "Informe o senha")]
        [MaxLength(50, ErrorMessage = "A senha pode ter no máximo 50 caracteres.")]
        public string Senha { get; set; }
        [Required(ErrorMessage = "Informe o nome")]
        [MaxLength(200, ErrorMessage = "O nome pode ter no máximo 200 caracteres.")]
        public string Nome { get; set; }
        [Required(ErrorMessage = "Informe o e-mail")]
        [MaxLength(200, ErrorMessage = "O e-mail pode ter no máximo 200 caracteres.")]
        public string Email {get;set; }
        public int IdPerfil { get; set; }
        public Perfil Perfil { get; set; }
        public KeyControle KeyC { get; set; }

        public override string ToString()
        {
            return "Id: " + Id + " | Login: " + Login + " | Senha: " + Senha
                + " | Nome: " + Nome + " | E-mail: " + Email + " | IdPerfil: " + IdPerfil;
        }

    }
}

File: mobile/lib/model/Usuario.dart
class Usuario {
  int id;
  String login;
  String senha;
  String nome;
  String email;
  int idPerfil;
  Perfil perfil;
  KeyC key;

  Usuario(
      {this.id,
        this.login,
        this.senha,
        this.nome,
        this.email,
        this.idPerfil,
        this.perfil,
        this.key});

  Usuario.fromJson(Map<String, dynamic> json) {
    id = json['Id'];
    login = json['Login'];
    senha = json['Senha'];
    nome = json['Nome'];
    email = json['Email'];
    idPerfil = json['IdPerfil'];
    perfil =
    json['Perfil'] != null ? new Perfil.fromJson(json['Perfil']) : null;
    key = json['KeyC'] != null ? new KeyC.fromJson(json['KeyC']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['Id'] = this.id;
    data['Login'] = this.login;
    data['Senha'] = this.senha;
    data['Nome'] = this.nome;
    data['Email'] = this.email;
    data['IdPerfil'] = this.idPerfil;
    if (this.perfil != null) {
      data['Perfil'] = this.perfil.toJson();
    }
    if (this.key != null) {
      data['KeyC'] = this.key.toJson();
    }
    return data;
  }


  @override
  String toString() {
    return "ID" + this.id.toString() + " - NOME: " + this.nome + " - EMAIL: " + this.email;
  }
}

class Perfil {
  int id;
  String nome;
  bool ativo;

  Perfil({this.id, this.nome, this.ativo});

  Perfil.fromJson(Map<String, dynamic> json) {
    id = json['Id'];
    nome = json['Nome'];
    ativo = json['Ativo'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['Id'] = this.id;
    data['Nome'] = this.nome;
    data['Ativo'] = this.ativo;
    return data;
  }
}

class KeyC {
  int id;
  String codigo;
  String criadaEm;
  String ultimoUso;
  int quantidadeDeChamadas;
  bool ativada;

  KeyC(
      {this.id,
        this.codigo,
        this.criadaEm,
        this.ultimoUso,
        this.quantidadeDeChamadas,
        this.ativada});

  KeyC.fromJson(Map<String, dynamic> json) {
    id = json['Id'];
    codigo = json['Codigo'];
    criadaEm = json['CriadaEm'];
    ultimoUso = json['UltimoUso'];
    quantidadeDeChamadas = json['QuantidadeDeChamadas'];
    ativada = json['Ativada'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['Id'] = this.id;
    data['Codigo'] = this.codigo;
    data['CriadaEm'] = this.criadaEm;
    data['UltimoUso'] = this.ultimoUso;
    data['QuantidadeDeChamadas'] = this.quantidadeDeChamadas;
    data['Ativada'] = this.ativada;
    return data;
  }

}